models/Document/PageSnippet.php

```diff
@@ -503,7 +503,11 @@ public function hasEditable(string $name)
     public function getEditables(): array
     {
         if ($this->editables === null) {
-            $this->setEditables($this->getDao()->getEditables());
+            $documentEditables = $this->getDao()->getEditables();
+            if ($this->supportsContentMaster() && $contentMasterEditables = $this->getContentMasterDocument()?->getDao()->getEditables()){
```

Comment:
I am personally not a fan of assignments in conditions. In Java or C+ this would work, but would have an unexpected outcome since the compiler would just check if the assignment works, and this would always return true.
Maybe wen can split it up to make it even more clear?